### PR TITLE
Fix replacment of lower cased "Unreleased" link references

### PR DESCRIPTION
**Pull request by markuspoerschke**
When a "Unreleased" link reference was lower cased, then it was not replaced. In Markdown, link references are case insensitive. Prettier for example will make them all lower case. Therefore the identifier rather than the label must be compared to find the "Unreleased" link reference.

### CODE REVIEW

**Comment by markuspoerschke** [comment]
As an example you can see this commit: https://github.com/markuspoerschke/extractum/commit/ce4bd666340957e2c16785e51b33a8d02d6b40cd

File: __tests__/fixtures/lowercase_link_reference/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+-   Reworked the login system. You have to provide a password now!
+
+## [0.1.0] - 2019-09-05
+
+### Added
+
+-   Initial release :tada:
+
+[unreleased]: https://github.com/foo/bar/compare/0.2.0...HEAD
```
Comment (markuspoerschke) [inline __tests__/fixtures/lowercase_link_reference/CHANGELOG.md:27]: The only difference to the "standard" test is the lower cased "u" from the word "unreleased".

**Review by thomaseizinger — APPROVED** [review]
Thank you!

I assume you created dist.js by running `yarn build`?

**Comment by markuspoerschke** [comment]
@thomaseizinger Yes, I created the `dist.js`, because the Git Commit Hook warned me about doing so.

**Pull request merged**